ui/src/components/Tasks/AddSample.js

```diff
@@ -1,138 +1,138 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { connect } from 'react-redux';
-import { reduxForm } from 'redux-form';
-import { Modal, ButtonToolbar, Button, Form } from 'react-bootstrap';
-import { InputField, FieldsRow } from './fields';
-import validate from './validate_add_sample';
+import { Modal, ButtonToolbar, Button, Form, Row, Col } from 'react-bootstrap';
+import { bindActionCreators } from 'redux';
+import { addSamplesToList } from '../../actions/sampleGrid';
+import { addSampleAndMount, addSamplesToQueue } from '../../actions/queue';
+import { showList } from '../../actions/queueGUI';
+import { useLocation } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
 
-class AddSample extends React.Component {
-  constructor(props) {
-    super(props);
+const REQUIRED_MSG = 'This field is required';
+const PATTERN = /^[\w+:-]*$/u;
+const PATTERN_MSG = 'Characters allowed: A-Z a-z 0-9 _+:-';
 
-    this.addAndEnqueue = this.addAndEnqueue.bind(this);
-    this.addAndMount = this.addAndMount.bind(this);
-    this._addAndEnqueue = this._addAndEnqueue.bind(this);
-    this._addAndMount = this._addAndMount.bind(this);
-    this.handleCancel = this.handleCancel.bind(this);
-    this.getDefaultSampleData = this.getDefaultSampleData.bind(this);
-    this.handleKeyPress = this.handleKeyPress.bind(this);
-  }
+function getSampleData(params) {
+  return {
+    ...params,
+    type: 'Sample',
+    defaultPrefix: `${params.proteinAcronym}-${params.sampleName}`,
+    location: 'Manual',
+    loadable: true,
+    tasks: [],
+  };
+}
 
-  getDefaultSampleData(params) {
-    let prefix = params.sampleName ? params.sampleName : 'noname';
+function AddSample(props) {
+  const {
+    show,
+    hide,
+    addSamplesToList,
+    addSampleAndMount,
+    addSamplesToQueue,
+    showList,
+  } = props;
 
-    if (params.proteinAcronym && params.sampleName) {
-      prefix = `${params.proteinAcronym}-${prefix}`;
-    }
+  const { register, formState, handleSubmit, setFocus } = useForm();
+  const { isSubmitted, errors } = formState;
 
-    return {
-      ...params,
-      type: 'Sample',
-      defaultPrefix: prefix,
-      location: 'Manual',
-      loadable: true,
-      tasks: [],
-    };
-  }
+  const { pathname } = useLocation();
 
-  handleCancel() {
-    this.props.hide();
-  }
-
-  _addAndEnqueue(params) {
-    const sampleData = this.getDefaultSampleData(params);
-    this.props.addToQueue(sampleData);
-    this.props.hide();
-  }
-
-  addAndEnqueue() {
-    this.props.handleSubmit(this._addAndEnqueue)();
-  }
+  useEffect(() => {
+    if (show) {
+      // Timeout required when creating new sample from "Samples" page
+      setTimeout(() => setFocus('sampleName'), 0);
```

Comment:
The previous solution with the `autoFocus` prop/attribute was not quite working.